modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -168,13 +168,6 @@ export class {{classname}} {
         if (typeof value === "object") {
             if (Array.isArray(value)) {
                 (value as any[]).forEach( elem => httpParams = this.addToHttpParamsRecursive(httpParams, elem, key));
-            } else if (value instanceof Date) {
```

Comment:
@amakhrov how about we leave this code here for users that manually provide the current type mapping to Date as a fallback?

Reply:
I still fail to understand how using DateTime=Date mapping can possibly work for anyone.
It emits Models with properties of type Date, while these are in fact strings - enforcing developers to hack it around (like `myModel.dateField as any as string`).

We can probably keep this till the major release (5.0) and clean it up then - what do you think?

Reply:
currently `DateTime=Date` can only work if no dates are used in any body param model, but only in query params

Reply:
I think 
https://github.com/OpenAPITools/openapi-generator/pull/5314/files#diff-5a152f3888a8d2712c9824bc1eff5db6L89
should be released with `4.3.0`
while
https://github.com/OpenAPITools/openapi-generator/pull/5314/files#diff-f61af8a4c4eb2720bfb15259e0dca419L155
should be reverted to allow a fallback for backwards compatibility. Moreover, this will be useful when a full date conversion is implemented later.
Leaving it here does not have an impact for the new default `DateTime=string` mapping.

Reply:
and the https://github.com/OpenAPITools/openapi-generator/pull/5314/files#diff-d270de95051055e6c877f904c360ecadL117-L128 part should also be kept, for backwards compatibility. it is checked for `instanceof Date`

Reply:
I'm ok with keeping the mapping in templates as is - although I believe the only reason it was ever needed is because the generated code (with DateTime=Date) forced users to pass Dates as params.
Now, if someone still wants to have this mapping for whatever reason (provided via --type-mapping argument), they can support that with a customized template (via --template-dir).

Reply:
A discarded the commits changing the templates - only keep the ones that change teh default typemapping